src/artifact_providers/__tests__/base.test.ts

```diff
@@ -9,33 +9,73 @@ describe('parseFilterOptions', () => {
     expect(parsedFilters).not.toHaveProperty('excludeNames');
   });
 
-  test('undefined properties', () => {
+  test.each([
+    [undefined, undefined],
+    [undefined, '/onlyExclude/'],
+    ['/onlyInclude/', undefined],
+    ['/include/', '/exclude/'],
+  ])('undefined and string properties', (includeNames, excludeNames) => {
     const rawFilters: RawFilterOptions = {
-      includeNames: undefined,
-      excludeNames: undefined,
+      includeNames: includeNames,
+      excludeNames: excludeNames,
     };
     const parsedFilters = parseFilterOptions(rawFilters);
-    expect(parsedFilters).not.toHaveProperty('includeNames');
-    expect(parsedFilters).not.toHaveProperty('excludeNames');
+
+    if (includeNames) {
+      expect(parsedFilters.includeNames).toStrictEqual(
+        stringToRegexp(includeNames)
+      );
+    } else {
+      expect(parsedFilters.includeNames).not.toBeDefined();
+    }
+    if (excludeNames) {
+      expect(parsedFilters.excludeNames).toStrictEqual(
+        stringToRegexp(excludeNames)
+      );
+    } else {
+      expect(parsedFilters.excludeNames).not.toBeDefined();
+    }
   });
 
-  test('string properties', () => {
-    const stringFilter = '/testFilter/';
+  test.each([
+    [undefined, undefined],
+    [undefined, new RegExp('onlyExclude')],
+    [new RegExp('onlyInclude'), undefined],
+    [new RegExp('include'), new RegExp('exclude')],
+  ])('undefined and regex properties', (includeNames, excludeNames) => {
```

Comment:
I think you can fold these cases into the one above (the `undefined` ones are already redundant) and add a typecheck to the expectation:

```ts
    expect(parsedFilters.includeNames).toBeInstanceOf(RegExp);
```

Reply:
Merging them would mean adding another type check before each expectation since having a string or regexp in the input is not the same. Separating them into two different test cases makes it easier to read and also to identify the failing test case (if any).